### PR TITLE
Fix: [Capacitor] Top safe area is covered

### PR DESCRIPTION
**Pull request by msdewitt**
#2514

This is a small fix for the overlap issue caused on IOS without the contentInset being set. By default, the app will overlap without this configuration.

Screenshots after fix on ios iPhone 15 emulator:

Screenshot 2024-11-06 at 12 35 26 PM

Screenshot 2024-11-06 at 12 36 57 PM

### CODE REVIEW

**Review by trevinhofmann — CHANGES_REQUESTED** [review]
Thank you, @msdewitt!

This change works when the user is at the top of the page, but the toolbar area is still hidden after scrolling down.

## iOS Simulator:

https://github.com/user-attachments/assets/f3615e6c-2e30-4e4f-9aab-dd07ded238f5

## Physical iOS:

https://github.com/user-attachments/assets/88933041-83ab-4d87-b6b9-c627f0081246

**Comment by msdewitt** [comment]
@trevinhofmann I'm not sure if that issue is covered in the bug card attached to this fix. This fix is only to fix the overlap of the top bar icon interacting with the phone icons. Do you have a link to the bug issue for the navbar not sticking to the top that you mentioned above? Thanks

**Comment by msdewitt** [comment]
@raineorshine I see now. I was confused because I thought the tool bar was not supposed to scroll with the movement, but I see it hiding after it scrolls in the video from watching it a few times. This is indeed an issue. Let me see what I can do.

**Comment by raineorshine** [comment]
Thanks! The toolbar is position fixed and expected to not scroll (i.e. it should stay at the top of the viewport regardless of the scroll position). For reference, it should behave the same as on desktop or Mobile Safari.

**Comment by msdewitt** [comment]
There is an update for this fix now which can handle the scrolling.

https://github.com/user-attachments/assets/4901bbfd-d737-4be0-9e0f-d2fe67c76a34


Please review when you can @trevinhofmann  and @raineorshine

**Comment by trevinhofmann** [comment]
Thank you, @msdewitt! I am away from my office at the moment, but I will be able to test the latest changes with physical devices tomorrow.

**Comment by raineorshine** [comment]
Thanks! That seems to fix it for the toolbar at least.

However, the content is still overlapping with the safe area when you scroll up. This is a frame from your video:

image

No part of the UI should overlap with the safe area, including the toolbar, menu, editor, scroll zone, footer, etc. The safe area should just be the background color, otherwise it interferes with the clock, connection, and battery icons that are rendered by iOS.

**Comment by msdewitt** [comment]
@raineorshine Actually, its not that it overlaps, but that by default the ios status bar will change from a solid color to transparent when you scroll down. This is a default behavior which we can change by using Capacitor StatusBar plugin. It's just that those changes overlap with another issue I am working on and was preparing to merge later. Do you want me to handle the Capacitor StatusBar changes in this fix or in this [issue](https://github.com/cybersemics/em/issues/1700)?

File: index.html [hunk 1]
```diff
@@ -24,7 +24,7 @@
     <title>em</title>
 
   </head>
-  <body>
+  <body style="padding-top: env(safe-area-inset-top);">
```
Comment (raineorshine) [inline index.html:27]: Let's apply this style in [panda.config.ts](https://github.com/cybersemics/em/blob/82c1f68a31df4086867f3e809c563cdfe2e84ac4/panda.config.ts#L159) for consistency.

**Review by raineorshine — CHANGES_REQUESTED** [review]
I see! Thanks for clarifying. Let's add the StatusBar plugin in a separate PR.

File: index.html (hunk 1 above)
Comment (msdewitt) [reply index.html]: This is actually unused and I removed it in the latest commit. Its good to go now.

**Comment by trevinhofmann** [comment]
@msdewitt -- Can you please have a look at the commits in this PR and get those cleaned up? It seems that there was some issue with a rebase/merge, because there are now many unrelated commits and file changes. Thank you!

**Comment by msdewitt** [comment]
@trevinhofmann Yes, I got too annoyed with the rebase on this branch and just made a new pr after closing this one. Please see: [Link](https://github.com/cybersemics/em/pull/2593) for further discussions. Its an easier review now.

**Pull request closed without merging**